keras_cv/layers/preprocessing/random_color_degeneration.py

```diff
@@ -55,7 +55,7 @@ def __init__(
         self.seed = seed
 
     def get_random_transformation(self, **kwargs):
-        return self.factor()
+        return self.factor(dtype=kwargs["image"].dtype)
```

Comment:
Everywhere where we're making this change -- instead of using `kwargs["image"]` let's just add `image=None` to the method signature

Reply:
+1